closed/src/java.base/share/native/libjli/criuhelpers.c

```diff
@@ -0,0 +1,309 @@
+ */
+static const char *
+getLogFile(int argc, char **argv, int *error)
+{
+    const char *logFile = NULL;
+    const char *logFilePropertyValue = getCommandLineOptionValue("-Dopenj9.internal.criu.logFile", argc, argv, error);
+    if (0 == *error) {
+        if (NULL != logFilePropertyValue) {
+            logFile = logFilePropertyValue;
+        } else {
+            JLI_ReportErrorMessage("The value of the command line option -Dopenj9.internal.criu.logFile was not found.");
+            *error = -2;
+        }
+    }
+    return logFile;
+}
+
+/**
+ * Restore the system state from a checkpoint using the CRIU tool.
+ * @param[in] checkpointDirectory The directory containing the checkpoint data
+ * @param[in] logLevel The log level for CRIU logging
+ * @param[in] unprivilegedModeOn Indicates whether the unprivileged mode option is on
+ * @param[in] logFile The log file option for CRIU
+ * @return 0 if the execution of the 'criu restore' command succeeds, -1 otherwise
+ */
+static int
+restoreFromCheckpoint(const char *checkpointDirectory, int logLevel, jboolean unprivilegedModeOn, const char *logFile)
+{
+    int restoreStatus = 0;
+    int length = -1;
+    char logLevelOption[4] = { 0 };
+    char *logFileOption = NULL;
+    int argc = 0;
+    const char *argv[9] = { NULL };
+    argv[argc++] = "criu";
+    argv[argc++] = "restore";
+    argv[argc++] = "-D";
+    argv[argc++] = checkpointDirectory;
+    length = snprintf(NULL, 0, "-v%d", logLevel);
+    if (length < 0) {
+        JLI_ReportErrorMessage("Failed to calculate the length of option '-v%d'.", logLevel);
+        restoreStatus = -1;
+        goto done;
+    }
+    if (snprintf(logLevelOption, sizeof(logLevelOption), "-v%d", logLevel) < 0) {
+        JLI_ReportErrorMessage("Failed to format the log level option '-v%d'.", logLevel);
+        restoreStatus = -1;
+        goto done;
+    }
+    argv[argc++] = logLevelOption;
+    argv[argc++] = "--shell-job";
+    if (unprivilegedModeOn) {
+        argv[argc++] = "--unprivileged";
+    }
+    if (NULL != logFile) {
+        length = strlen(logFile) + sizeof("--log-file=%s") - 1;
+        logFileOption = (char *)JLI_MemAlloc(length - 1);
+        if (NULL == logFileOption) {
+            JLI_ReportErrorMessage("Failed to allocate memory for option '--log-file=%s'.", logFile);
+            restoreStatus = -1;
+            goto done;
+        }
+        if (snprintf(logFileOption, length - 1, "--log-file=%s", logFile) < 0) {
+            JLI_ReportErrorMessage("Failed to format option '--log-file=%s'.", logFile);
+            restoreStatus = -1;
+            goto done;
+        }
+        argv[argc++] = logFileOption;
+    }
+    argv[argc] = NULL;
+    execvp(argv[0], (char * const *)argv);
+    /* If execvp returns, there was an error. */
+    restoreStatus = -1;
+freeLogFileOption:
+    if (NULL != logFileOption) {
+        JLI_MemFree((void *)logFileOption);
+        logFileOption = NULL;
```

Comment:
This assignment is not useful (there are no following reads).

Reply:
Removed it.